api-spec.yaml

```
@@ -0,0 +1,482 @@
+openapi: 3.0.3
+
+info:
+  title: Eris Library - OpenAPI 3.0
+  description:
+    The open source and scalable project for manage public library
+  license:
+    name: GNU GENERAL PUBLIC LICENSE Version 3, 29 June 2007
+    url: https://www.gnu.org/licenses/gpl-3.0.en.html
+  version: 1.0.0
+servers:
+  - url: localhost:8080
+tags:
+  - name: books
+    description: Everything about books
+  - name: members
+    description: members information
+  - name: loans
+
+paths:
+  /books:
+    put:
+      tags:
+        - books
+      summary: Update an existing book
+      operationId: updateBook
+      requestBody:
+        description: Update an existent book
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Book'
+
+        required: true
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: book not found
+        '405':
+          description: Validation exception
+      security:
+        - library_auth:
+            - write:books
+            - read:books
+    post:
+      tags:
+        - books
+      summary: Add a new book to the library
+      operationId: addBook
+      requestBody:
+        description: Create a new book in the library
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Book'
+        required: true
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+
+        '405':
+          description: Invalid input
+      security:
+        - library_auth:
+            - write:books
+            - read:books
+    get:
+      summary: Search for books by title and/or author
+      description: Returns a list of books matching the search criteria
+      parameters:
+        - in: query
+          name: title
+          schema:
+            type: string
+          description: Title of the book to search for
+        - in: query
+          name: author
+          schema:
+            type: string
+          description: Author of the book to search for
+      responses:
+        '200':
+          description: A list of books matching the search criteria
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
```

A response is an array:

```suggestion
              schema:
                type: array
                  items:
                    $ref: '#/components/schemas/Book'
```